habushu-maven-plugin/src/main/java/org/technologybrewery/habushu/ContainerizeDependenciesMojo.java

```diff
@@ -0,0 +1,156 @@
+public class ContainerizeDependenciesMojo extends AbstractHabushuMojo {
+
+    private static final Logger logger = LoggerFactory.getLogger(ContainerizeDependenciesMojo.class);
+
+    @Component
+    protected MavenSession session;
+
+    protected File anchorSourceDirectory;
+    protected File anchorOutputDirectory;
+    protected final String HABUSHU = "habushu";
+
+    /**
+     * Overriding to allow execution in non-habushu projects.
+     */
+    @Override
+    public void execute() throws MojoExecutionException, MojoFailureException {
+        doExecute();
+    }
+
+    @Override
+    protected void doExecute() throws MojoExecutionException, MojoFailureException {
+        if (this.anchorSourceDirectory == null) {
+            this.anchorSourceDirectory = new File(session.getExecutionRootDirectory());
+        }
+
+        Set<MavenProject> habushuProjects = getHabushuProjects();
+        try {
+            copyRelevant(habushuProjects);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * Copies the relevant source files by leveraging {@link FileSet}s to filter appropriately.
+     * @param habushuProjects corresponding projects of the pom's habushu-type dependencies
+     * @throws IOException
+     */
+    private void copyRelevant(Set<MavenProject> habushuProjects) throws IOException {
+        this.anchorOutputDirectory = new File(project.getBuild().getDirectory()
+                + "/venv-support/" + this.anchorSourceDirectory.getName());
+        Path srcRoot = this.anchorSourceDirectory.toPath();
+        Path destRoot = this.anchorOutputDirectory.toPath();
+
+        Map<Path, FileSet> dependencyFileSets = new HashMap<>();
+        for (MavenProject project : habushuProjects) {
+            Path projectPath = project.getFile().getParentFile().toPath();
+            Path relativeProjectPath = srcRoot.relativize(projectPath);
+            FileSet fileSet = getDefaultFileSet();
+            fileSet.setDirectory(projectPath.toString());
+            dependencyFileSets.put(relativeProjectPath, fileSet);
+        }
+
+        FileSetManager fileSetManager = new FileSetManager();
+        for (Path project : dependencyFileSets.keySet()) {
+            FileSet fileSet = dependencyFileSets.get(project);
+            logger.info("Copying {} files from {}",
+                    fileSetManager.getIncludedFiles(fileSet).length,
+                    project.getFileName()
+            );
+            for (String includedFile : fileSetManager.getIncludedFiles(fileSet)) {
+                Path relativePath = project.resolve(includedFile);
+                Files.createDirectories(destRoot.resolve(relativePath).getParent());
+                Files.copy(srcRoot.resolve(relativePath), destRoot.resolve(relativePath));
+            }
+        }
+    }
+
+    private FileSet getDefaultFileSet() {
+        FileSet fileSet = new FileSet();
+        fileSet.addExclude("target/**");
```

Comment:
S: `target`, `dist`, and `.venv` are all possible to override or be different.  We need to dynamically get the values.

Reply:
That will be covered in a follow-on to enable the user to pass in via configs. We just have it this way as the default, or else using it in the baseline will take a very long time, due to `.venv` in particular taking a while to copy

Reply:
Not sure I follow or agree.  You can get the dist directory by adding the appropriate member variable:

Reply:
Suggestion has been implemented. Please note that `findCurrentVirtualEnvironmentFullPath()` and `getCleanVirtualEnvironmentPath()` have been refactored to live in `HabushuUtil`. `findCurrentVirtualEnvironmentFullPath()` now accepts as arguments, the various parameters it got from the `CleanHabushuMojo`. Necessary parameters have also been added to the `ContainerizeDepsMojo` as well.